worker/runtime/process_killer.go

```diff
@@ -70,10 +74,18 @@ func (p processKiller) Kill(
 		}
 
 		return fmt.Errorf("waitctx done: %w", err)
-	case status := <-statusC:
-		err = status.Error()
+	case exitStatus := <-procWaitStatus:
+		err = exitStatus.Error()
 		if err != nil {
-			return fmt.Errorf("waiting for exit status: %w", err)
+			grpcErr := status.Convert(err)
```

Comment:
Is it possible for `err` to not be a gRPC error? Should we use `grpcErr, ok := status.FromError(err)` instead and handle the case where `!ok`?

Reply:
@aoldershaw When `ok` is `false` the `FromError()` returns a `Status` with `codes.Unknown`. So the `DeadlineExceeded` condition will get skipped and the original `err` object will get returned.

Reply:
makes sense, thanks for the explanation!